src/patternfly/components/InputGroup/input-group.scss

```diff
@@ -31,13 +31,82 @@
   // Input group text, disabled variant
   --#{$input-group}__item--m-disabled__text--Color: var(--pf-t--global--text--color--on-disabled);
   --#{$input-group}__item--m-disabled--BackgroundColor: var(--pf-t--global--background--color--disabled--default);
+
+  // Input group search expanded
+  --#{$input-group}-expandable__search-input--TransitionTimingFunction--expand: var(--pf-t--global--motion--timing-function--decelerate);
+  --#{$input-group}-expandable__search-input--TransitionTimingFunction--collapse: var(--pf-t--global--motion--timing-function--accelerate);
+  --#{$input-group}-expandable__search-input--TransitionDuration--expand: var(--pf-t--global--motion--duration--fade--default);
+  --#{$input-group}-expandable__search-input--TransitionDuration--collapse: var(--pf-t--global--motion--duration--fade--short);
+  --#{$input-group}-expandable__search-input--ScaleX: 1;
+  --#{$input-group}-expandable__search-input--TransformOriginX--expand-left: 100%;
+  --#{$input-group}-expandable__search-input--TransformOriginX--expand-right: 0;
+  --#{$input-group}-expandable__search-input--TransformOriginX: var(--#{$input-group}-expandable__search-input--TransformOriginX--expand-right);
+  --#{$input-group}-expandable__button-icon--TransitionDuration--Opacity: var(--pf-t--global--motion--duration--fade--default);
+  --#{$input-group}-expandable__button-icon--TransitionTimingFunction--Opacity: var(--pf-t--global--motion--timing-function--default);
+  
+  @media (prefers-reduced-motion: no-preference) {
+    --#{$input-group}-expandable__search-input--ScaleX: .7;
+  }
 }
 
 .#{$input-group} {
   display: flex;
   gap: var(--#{$input-group}--Gap);
   width: 100%;
 }
+  
+.#{$input-group}-expandable {
+  &:not(.pf-m-expanded) {
+    .#{$pf-prefix}c-search-input {
+      display: none;
+    }
+  }
+
+  .#{$pf-prefix}c-search-input {
+    width: 100%;
+    opacity: 0;
+    transition-timing-function: var(--#{$input-group}-expandable__search-input--TransitionTimingFunction--collapse);
+    transition-duration: var(--#{$input-group}-expandable__search-input--TransitionDuration--collapse);
+    transition-property: opacity, scale, display;
```

Comment:
Instead of transitioning `display`, can you transition `visibility` and `width` to hide it? Then it will work x-browser without needing `allow-discrete` and `@starting-style`